src/Index/Index.php

```diff
@@ -423,6 +426,9 @@ private function indexDefinition(int $level, array $parts, array &$storage, Defi
      */
     private function removeIndexedDefinition(int $level, array $parts, array &$storage, array &$rootStorage)
     {
+        if (empty($parts)) {
```

Comment:
I don't really understand this function well enough to be confident about this code, but it fixes an exception that I got before and it seems intuitively right to have a base case for the recursion.